src/storage/src/store.rs

```diff
@@ -192,8 +195,10 @@ pub trait StateStore: Send + Sync + 'static + Clone {
     /// Otherwise, only data of the provided epoch will be synced.
     fn sync(&self, epoch: u64) -> Self::SyncFuture<'_>;
 
+    fn await_sync_epoch(&self, epoch: u64) -> Self::AwaitSyncEpochFuture<'_>;
```

Comment:
Btw, I think we can keep the original `sync` method since by convention `sync` already means waiting for data to be flushed to disk. It is okay to address this in your next PR.

Reply:
This PR meant to decouple `sync` into `seal_epoch` and `await_sync_epoch`. The `sync` is kept and has the implementation as
```
fn sync(&self, epoch: u64) -> Self::SyncFuture<'_> {
        async move {
            self.seal_epoch(epoch, true);
            self.await_sync_epoch(epoch).await
        }
    }
```

Reply:
I mean we don't need to keep both `sync` and `await_sync_epoch` after the refactoring is fully done.

Reply:
Providing a method to wait for a sealed epoch to finish syncing is necessary, so `await_sync_epoch` should be kept.

`sync` is more than just awaiting for an epoch to be synced. It also includes sealing an epoch. Our release code will not contain usage of `sync` after this PR, but `sync` is still widely used in our unit tests. If we remove `sync`, we will have to replace all the current `sync` into `seal_epoch` + `await_sync_epoch`. A feasible way to do this is remove `sync` from the `StateStore` trait, and introduce `sync` as a method for unit tests for `HummockStorage` only, though it looks weird if our kv-like `StateStore` trait does not provide a `sync` method.